### PR TITLE
Fix filter return value

### PR DESCRIPTION
**Pull request by bperian**
Filter always returned true

### CODE REVIEW

File: lib/logstash/filters/elasticsearch.rb [hunk 1]
```diff
@@ -121,7 +121,7 @@ def filter(event)
       @logger.warn("Failed to query elasticsearch for previous event", :index => @index, :query => query, :event => event, :error => e)
       @tag_on_failure.each{|tag| event.tag(tag)}
     end
-    filter_matched(event)
+    filter_matched(event) if !resultsHits.nil? && !resultsHits.empty?
```
Comment (jsvd) [inline lib/logstash/filters/elasticsearch.rb]: not sure if this is the ideal criteria. First, we also have [the result of aggregations](https://github.com/bperian/logstash-filter-elasticsearch/blob/cc6455d28217b560db7ee96f0313766c49a4334e/lib/logstash/filters/elasticsearch.rb#L113) which may not be empty. If an aggregation is requested with `size: 0` then no hits will be returned, only aggregations. Maybe we should do something like:

```ruby
# check if the hits array exists and if there are hits
# if the query is an aggregation using the query dsl with size 0 we'll still see total hits > 0
filter_matched(event) if (resultsHits && results["hits"]["total"] > 0)
```

Comment (jsvd) [inline lib/logstash/filters/elasticsearch.rb]: I also suggest placing this in the `else` clause of this rescue block:

```ruby
def filter(event)
  begin
    # perform query
    # process results
  rescue
    # tag as error
  else
    # filter_matched if query returned hits
  end
end
```

Comment (bperian) [reply lib/logstash/filters/elasticsearch.rb]: Thank you for the feedback. The lack of that condition affects any action that is based on the result of filter matched ( tagging, fields, etc. ) from my tests. I'll get back with changes asap that will acommodate all feedback ( and also do thorough testing)

**Comment by bperian** [comment]
It has to pass Travis, right? I can't reproduce it. Will do some more testing

**Comment by jsvd** [comment]
@bperian thanks for the contribution so far! any chance I can persuade you to add tests? I linked before [to existing examples](https://github.com/logstash-plugins/logstash-filter-elasticsearch/blob/master/spec/filters/elasticsearch_spec.rb#L112-L125) you could use to test that:

a) a successful query results in events with the correct tag of a filter_matched
b) an empty query doesn't add the tag of a filter_matched

**Comment by bperian** [comment]
Of course. Will get back with the tests

**Comment by webmat** [comment]
Hi @bperian, I'll be spending some time on this plugin in the next little while.

If you're able to submit these tests soon, I'd gladly merge this in.

**Comment by bperian** [comment]
I added two tests for tagging
1. Tagging expected behaviour for query with results 
2. Tagging expected behaviour for query with no results

**Comment by webmat** [comment]
Closing. Version 3.3.1 is published and fixes this.

Once again, thanks for your contribution, @bperian!

**Pull request closed without merging**